### PR TITLE
feat: customize MaxRecurringJobRetain

### PR DESCRIPTION
**Pull request by mantissahz**
Add a global setting to allow users to customize the maximum retain number of the snapshot or backup for a recurring job.

This number should not exceed the maximum number of snapshots (250).

Ref: longhorn/longhorn#5713

### CODE REVIEW

File: types/setting.go [hunk 1]
```diff
@@ -94,6 +97,7 @@ const (
 	SettingNameFailedBackupTTL                                          = SettingName("failed-backup-ttl")
 	SettingNameRecurringSuccessfulJobsHistoryLimit                      = SettingName("recurring-successful-jobs-history-limit")
 	SettingNameRecurringFailedJobsHistoryLimit                          = SettingName("recurring-failed-jobs-history-limit")
+	SettingNameRecurringJobMaxRetain                                    = SettingName("recurring-job-max-retain")
```
Comment (innobead) [inline types/setting.go:100]: Retain is verb, let's use `retention` instead. Please change the var name and description.

File: types/setting.go [hunk 2]
```diff
@@ -956,6 +962,16 @@ var (
 		Default:  "1",
 	}
 
+	SettingDefinitionRecurringJobMaxRetain = SettingDefinition{
+		DisplayName: "Recurring Job Maximum Retain Number",
```
Comment (innobead) [inline types/setting.go:966]: Change the display name to `Maximum Retention Number for Recurring Job`.

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Just two comments need to resolve first.

File: types/setting.go (hunk 2 above)
Comment (mantissahz) [reply types/setting.go]: Modified. Thanks

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mantissahz Make sure we have a PR to update longhorn/longhorn repo as well.

**Pull request merged**